tests/core/search/test_search_filter.py

```diff
@@ -144,7 +305,15 @@ def test_parse_malformedString_fails(self):
             (')', ')'),
             ('()', '()'),
             ('a:1 AND (b:2 OR (c:3 AND (d:4 OR e:5)) OR d:6', 'imbalanced parentheses, not closed'),
-            ('a:1 AND b:2 OR (c:3 AND (d:4 OR e:5))) OR d:6', 'imbalanced parentheses, not opened')
+            ('a:1 AND b:2 OR (c:3 AND (d:4 OR e:5))) OR d:6', 'imbalanced parentheses, not opened'),
+
+            # In term tests
```

Comment:
I can think of more edge cases like `[ ]` inside `( )`. Use your judgement to see if these cases should be tested or not. Maybe chatgpt can help you generate these edge cases quickly.

Reply:
Added an edge case! Can't think of any other meaningful tests for this.